addons/stock/models/stock_quant.py

```diff
@@ -463,6 +463,22 @@ def action_inventory_at_date(self):
             "context": context,
         }
 
+    def action_stock_quant_relocate(self):
+        if len(self.company_id) > 1 or not all(q.company_id.id for q in self) or not all(q > 0 for q in self.mapped('quantity')):
+            raise UserError(_('You can only move positive quantities stored in a single company.\n'
+                              'You cannot move multiple quantities from different companies '
+                              'or quantities in a location not belonging to any company or null/negative quantities.'))
```

Comment:
no, we leave that to inter-company flows. Relocate is purely for internal moves

Reply:
I worded that poorly. I mean we can move a bunch quants within company_1 and then a bunch of quants within company_2, we just have to do it in 2 action calls rather than in 1

Reply:
ah, in that case yes, it's how it's supposed to be done